etc/ci.sh

```diff
@@ -46,10 +48,23 @@ make
 # return to base directory
 popd
 
-# create dir for coverage results
-COVDIR=coverage
-mkdir -p $COVDIR
-
+# When we want to record coverage, we enable the user preference
+# ReproducibleBehaviour, which stops code depending on Runtime(),
+# which give different results on different test executions.
+if [ -z "${NO_COVERAGE+x}" ]
+then
```

Comment:
Hmmm.. wouldn't it be either to do something like this?
```sh
if [[ -n ${NO_COVERAGE} ]]
then
    bin/gap.sh tst/${TEST_SUITE}.g
    exit 0
fi
```
and then get rid of the other changes below?